packages/dev/core/src/Meshes/lattice.material.ts

```diff
@@ -0,0 +1,302 @@
+                vec3d p010 = texelFetch(latticeData, ivec3c(ii0, jj1, kk0), 0).rgb;
+                vec3d p110 = texelFetch(latticeData, ivec3c(ii1, jj1, kk0), 0).rgb;
+                vec3d p001 = texelFetch(latticeData, ivec3c(ii0, jj0, kk1), 0).rgb;
+                vec3d p101 = texelFetch(latticeData, ivec3c(ii1, jj0, kk1), 0).rgb;
+                vec3d p011 = texelFetch(latticeData, ivec3c(ii0, jj1, kk1), 0).rgb;
+                vec3d p111 = texelFetch(latticeData, ivec3c(ii1, jj1, kk1), 0).rgb;
+
+                // Trilinear interpolation
+                vec3d p00 = mix(p000, p100, tx);
+                vec3d p01 = mix(p001, p101, tx);
+                vec3d p10 = mix(p010, p110, tx);
+                vec3d p11 = mix(p011, p111, tx);
+
+                vec3d p0 = mix(p00, p10, ty);
+                vec3d p1 = mix(p01, p11, ty);
+
+                vec3d deformedPos = mix(p0, p1, tz);
+                positionUpdated = deformedPos + lattice_position;
+            };
+        `;
+
+        if (shaderLanguage === ShaderLanguage.WGSL) {
+            code =
+                `
+                let lattice_min = uniforms.lattice_min;
+                let lattice_max = uniforms.lattice_max;
+                let lattice_resolution = uniforms.lattice_resolution;
+                let lattice_position = uniforms.lattice_position;
+                let lattice_cellSize = uniforms.lattice_cellSize;
+            ` + code;
+
+            code = code.replace(/ivec3c/g, "vec3i");
+            code = code.replace(/vec3d/g, "var");
+            code = code.replace(/vec3c/g, "vec3f");
+            code = code.replace(/intd/g, "var");
+            code = code.replace(/intc/g, "i32");
+            code = code.replace(/floatd/g, "var");
+            code = code.replace(/floatc/g, "f32");
+            code = code.replace(/texelFetch/g, "textureLoad");
+        } else {
+            code = code.replace(/ivec3c/g, "ivec3");
+            code = code.replace(/vec3d/g, "vec3");
+            code = code.replace(/vec3c/g, "vec3");
+            code = code.replace(/intd/g, "int");
+            code = code.replace(/intc/g, "int");
+            code = code.replace(/floatd/g, "float");
+            code = code.replace(/floatc/g, "float");
+        }
+        this._code = code;
+
+        return this._code;
+    }
+
+    /**
+     * Returns a list of custom shader code fragments to customize the shader.
+     * @param shaderType "vertex" or "fragment"
+     * @param shaderLanguage The shader language to use.
+     * @returns null if no code to be added, or a list of pointName =\> code.
+     */
+    public override getCustomCode(shaderType: string, shaderLanguage = ShaderLanguage.GLSL) {
+        if (shaderType === "vertex") {
+            // we're adding this specific code at the end of the main() function
+            if (shaderLanguage === ShaderLanguage.WGSL) {
+                return {
+                    CUSTOM_VERTEX_DEFINITIONS: `

```

Comment:
wondering if we should use this or add new injection points ? or move back the injections points in the new code :-)